src/transformers/models/pop2piano/tokenization_pop2piano.py

```diff
@@ -0,0 +1,794 @@
+    },
+    "vocab_special_file": {
+        "sweetcocoa/pop2piano": "https://huggingface.co/sweetcocoa/pop2piano/blob/main/vocab_special.json",
+    },
+}
+
+
+def token_time_to_note(number, cutoff_time_idx, current_idx):
+    current_idx += number
+    if cutoff_time_idx is not None:
+        current_idx = min(current_idx, cutoff_time_idx)
+
+    return current_idx
+
+
+def token_note_to_note(number, current_velocity, default_velocity, note_onsets_ready, current_idx, notes):
+    if note_onsets_ready[number] is not None:
+        # offset with onset
+        onset_idx = note_onsets_ready[number]
+        if onset_idx < current_idx:
+            # Time shift after previous note_on
+            offset_idx = current_idx
+            notes.append([onset_idx, offset_idx, number, default_velocity])
+            onsets_ready = None if current_velocity == 0 else current_idx
+            note_onsets_ready[number] = onsets_ready
+    else:
+        note_onsets_ready[number] = current_idx
+    return notes
+
+
+class Pop2PianoTokenizer(PreTrainedTokenizer):
+    """
+    Constructs a Pop2Piano tokenizer. This tokenizer does not require training.
+
+    This tokenizer inherits from [`PreTrainedTokenizer`] which contains most of the main methods. Users should refer
+    to: this superclass for more information regarding those methods. However the code does not allow that and only
+    supports composing from various genres.
+
+    Args:
+        vocab_params_file (`str`):
+            Path to the tokenizer file which contains parameters for the tokenizer.
+        vocab_time_file (`str`):
+            Path to the tokenizer file which contains vocabulary for "TIME" tokens.
+        vocab_note_file (`str`):
+            Path to the tokenizer file which contains vocabulary for "NOTE" tokens.
+        vocab_velocity_file (`str`):
+            Path to the tokenizer file which contains vocabulary for "VELOCITY" notes.
+        vocab_special_file (`str`):
+            Path to the tokenizer file which contains vocabulary for "SPECIAL" tokens.
+    """
+
+    model_input_names = ["token_ids"]
+    vocab_files_names = VOCAB_FILES_NAMES
+    pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
+
+    def __init__(
+        self,
+        vocab_params_file,
+        vocab_time_file,
+        vocab_note_file,
+        vocab_velocity_file,
+        vocab_special_file,
+        unk_token="-1",
+        eos_token="1",
+        pad_token="0",
+        bos_token="2",
+        **kwargs,
+    ):
+        unk_token = AddedToken(unk_token, lstrip=False, rstrip=False) if isinstance(unk_token, str) else unk_token
+        eos_token = AddedToken(eos_token, lstrip=False, rstrip=False) if isinstance(eos_token, str) else eos_token
+        pad_token = AddedToken(pad_token, lstrip=False, rstrip=False) if isinstance(pad_token, str) else pad_token
+        bos_token = AddedToken(bos_token, lstrip=False, rstrip=False) if isinstance(bos_token, str) else bos_token
```

Comment:
Just to be sure: If I try to call `tokenizer.eos_token_id`, I'm probably gonna get `int(self.encoder["TOKEN_TIME_VOCAB"].get(token, self.unk_token))` ( since it calls with `token_type = 3`) which will return `135`. If this is expected then no problem!

Reply:
yes it is expected